### PR TITLE
Some doucment improvements

### PR DESCRIPTION
**Pull request by bwangYext**
Made some minor improvements:
- Reorded some sections.
- Added missing configuration options for components.
- Be more explicit about putting addComponent calls in onReady.
- Not required -> optional.
- Add link to API key documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,7 +115,7 @@ To start, every component requires an HTML container.
 <div class="search-container"></div>
 ```
 
-Then, you can add a component to your page through the ANSWERS add interface.
+Then, you can add a component to your page through the ANSWERS add interface. You need to call `addComponent` from `onReday`.
```
Comment (Delerme) [inline README.md]: typo

File: README.md [hunk 2]
```diff
@@ -323,43 +288,147 @@ Each provide a different way of auto complete.
 ANSWERS.addComponent('SearchBar', {
   container: '.search-query-container',
   title: 'Search my Brand',                 // optional, defaults to 'Answers'
+  query: 'query',                           // optional, the query string to use for the input box
```
Comment (Delerme) [inline README.md]: maybe clarify "initial query string..."

File: README.md [hunk 3]
```diff
@@ -323,43 +288,147 @@ Each provide a different way of auto complete.
...
   labelText: 'What are you looking for?',   // optional, defaults to 'What are you interested in?'
+  submitText: 'Submit',                     // optional, used for labeling the submit button, also provided to the template
+  submitIcon: 'path/to/icon',               // optional, an icon for the submit butto
```
Comment (Delerme) [inline README.md]: typo

File: README.md [hunk 4]
```diff
@@ -323,43 +288,147 @@ Each provide a different way of auto complete.
...
+  promptHeader: 'Header',                   // optioanl, shown as the first item for auto complete
```
Comment (Delerme) [inline README.md]: typo

File: README.md [hunk 5]
```diff
@@ -323,43 +288,147 @@ Each provide a different way of auto complete.
...
   placeholderText: 'Start typing...'        // optional, no default
   autoFocus: true,                          // optional, defaults to false
   autoCompleteOnLoad: false,                // optional, when auto focus on load, optionally open the autocomplete
   searchCooldown: 2000,                     // optional, defaults to 300ms (0.3 seconds)
   promptForLocation: true,                  // optional, asks the user for their geolocation when "near me" intent is detected
   clearButton: true,                        // optional, displays an "x" button to clear the current query when true
-  redirectUrl: 'path/to/url'                // optional, redirect search query to url
+  redirectUrl: 'path/to/url',               // optional, redirect search query to url
+  formSelector: 'form',                     // optional, defaults to native form node within container
+  inputEl: '.js-yext-query'                // optional, the input element used for searching and wires up the keyboard interaction
```
Comment (Delerme) [inline README.md]: comment spacing

File: README.md [hunk 6]
```diff
@@ -397,10 +466,51 @@ ANSWERS.addComponent('FilterBox', {
         }
       ]
     }
-  ]
+  ],
+  // Required, the vertical key for the search, default null
+  verticalKey: 'verticalKey',
+  // Optional, if triggers a search on each change to a filter, default false
```
Comment (Delerme) [inline README.md]: "if true..."

File: README.md [hunk 7]
```diff
@@ -441,7 +551,15 @@ ANSWERS.addComponent('FilterOptions', {
       field: 'c_storeType',
       value: 'Megastore'
     }
-  ]
+  ],
+  // Optional, the selector used for options in the template
+  optionSelector: '.js-option',
+  // Optional, if triggers a search on each change to a filter, default false
```
Comment (Delerme) [inline README.md]: "if true.."

File: README.md [hunk 8]
```diff
@@ -467,7 +585,11 @@ ANSWERS.addComponent('RangeFilter', {
   // The initial min value to show, defaults to 0
   initialMin: 1,
   // The initial max value to show, defaults to 10
-  initialMax: 5
+  initialMax: 5,
+  // Optional, if triggers a search on each change to a filter, default false
```
Comment (Delerme) [inline README.md]: "if true..."

File: README.md [hunk 9]
```diff
@@ -493,7 +615,13 @@ ANSWERS.addComponent('DateRangeFilter', {
   // The initial min date to show in yyyy-mm-dd format, defaults to today
   initialMin: '2019-08-01',
   // The initial max date to show in yyyy-mm-dd format, defaults to today
-  initialMax: '2019-09-01'
+  initialMax: '2019-09-01',
+  // If true, this filter represents an exclusive range, rather than an inclusive one
+  isExclusive: false,
+  // Optional, if triggers a search on each change to a filter, default false
```
Comment (Delerme) [inline README.md]: "if true..."

File: README.md [hunk 10]
```diff
@@ -508,129 +636,63 @@ Displays a "Use My Location" button that filters results to a radius around the
...
   // The label to show when unable to get the user's location
-  errorLabel: 'Unable To Use Location',
-  // CSS selector of the button
-  buttonSelector: '.js-yxt-GeoLocationFilter-button'
+  errorText: 'Unable To Use Location',
+  // Optional, CSS selector of the button
+  buttonSelector: '.js-yxt-GeoLocationFilter-button',
+  // OPtional, Css selector of the query input
+  inputSelector: '.js-yxt-GeoLocationFilter-input',
+  // Optional, if triggers a search on each change to a filter, default false
```
Comment (Delerme) [inline README.md]: "if true..."

File: README.md [hunk 11]
```diff
@@ -508,129 +636,63 @@ Displays a "Use My Location" button that filters results to a radius around the
...
+  searchOnChange: true,
+  // Optional, the icon url to show in the geo button
+  geoButtonIcon: 'path/to/url',
+  // Optional, the text toe show in the geo button
```
Comment (Delerme) [inline README.md]: typo

**Review by Delerme — COMMENTED** [review]
nice thanks

File: README.md (hunk 7 above)
Comment (bwangYext) [reply README.md]: Done.

File: README.md (hunk 1 above)
Comment (bwangYext) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (bwangYext) [reply README.md]: Done.

File: README.md (hunk 3 above)
Comment (bwangYext) [reply README.md]: Done.

File: README.md (hunk 5 above)
Comment (bwangYext) [reply README.md]: Done.

File: README.md (hunk 9 above)
Comment (bwangYext) [reply README.md]: Doen.

File: README.md (hunk 10 above)
Comment (bwangYext) [reply README.md]: Done.

File: README.md (hunk 11 above)
Comment (bwangYext) [reply README.md]: Done.

File: README.md (hunk 4 above)
Comment (bwangYext) [reply README.md]: Done.

File: README.md (hunk 8 above)
Comment (bwangYext) [reply README.md]: Done

**Review by b1lly — APPROVED** [review]

**Pull request merged**